File: admin/src/app/00_INFRA/types/Rating.ts
export type Rating = {
  id: string
  orderId: string
  reviewerId: string
  reviewerRole: "CUSTOMER" | "DRIVER"
  revieweeId: string
  revieweeRole: "CUSTOMER" | "DRIVER"
  rating: number
  comment?: string
  createdAt?: Date
  updatedAt?: Date
}

export type UserRatingStats = {
  averageRating: number
  totalRatings: number
  ratingDistribution: {
    1: number
    2: number
    3: number
    4: number
    5: number
  }
}

File: app/app/src/main/java/com/example/potiongo/data/Rating.kt
package com.example.potiongo.data

import com.google.firebase.Timestamp

data class Rating(
    val id: String = "",
    val orderId: String = "",
    val reviewerId: String = "",
    val reviewerRole: String = "",
    val revieweeId: String = "",
    val revieweeRole: String = "",
    val rating: Int = 0,
    val comment: String? = null,
    val createdAt: Timestamp? = null
)
